resourcemanager/recaser/recase.go

```diff
@@ -0,0 +1,96 @@
+package recaser
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/resourceids"
+)
+
+// ReCase tries to determine the type of Resource ID defined in `input` to be able to re-case it from
+func ReCase(input string) string {
+	return ReCaseWithIds(input, knownResourceIds)
+}
+
+// ReCaseWithIds tries to determine the type of Resource ID defined in `input` to be able to re-case it from based on an input list of Resource IDs
+func ReCaseWithIds(input string, ids map[string]resourceids.ResourceId) string {
```

Comment:
Can we make this function private/unexported? This should only be used for testing purposes, else we might end up using this unintentionally (which would be a bug, since it'd mean we're not registering all of the Resource IDs)

```suggestion
// reCaseWithIds tries to determine the type of Resource ID defined in `input` to be able to re-case it from based on an input list of Resource IDs
func reCaseWithIds(input string, ids map[string]resourceids.ResourceId) string {
```